packages/@aws-cdk/aws-codedeploy/lib/deployment-configuration.ts

```diff
@@ -0,0 +1,155 @@
+}
+
+/**
+ * Reference to a custom Deployment Configuration for an EC2/on-premise Deployment Group.
+ */
+export abstract class CustomServerDeploymentConfigurationRef extends cdk.Construct
+        implements ServerDeploymentConfiguration {
+    /**
+     * Import a custom Deployment Configuration for an EC2/on-premise Deployment Group defined either outside the CDK,
+     * or in a different CDK Stack and exported using the {@link #export} method.
+     *
+     * @param parent the parent Construct for this new Construct
+     * @param id the logical ID of this new Construct
+     * @param props the properties of the referenced custom Deployment Configuration
+     * @returns a Construct representing a reference to an existing custom Deployment Configuration
+     */
+    public static import(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationRefProps):
+            CustomServerDeploymentConfigurationRef {
+        return new ImportedCustomServerDeploymentConfigurationRef(parent, id, props);
+    }
+
+    public abstract readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public abstract readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    public export(): CustomServerDeploymentConfigurationRefProps {
+        return {
+            deploymentConfigurationName: new cdk.Output(this, 'DeploymentConfigurationName', {
+                value: this.deploymentConfigurationName,
+            }).makeImportValue(),
+        };
+    }
+}
+
+class ImportedCustomServerDeploymentConfigurationRef extends CustomServerDeploymentConfigurationRef {
+    public readonly deploymentConfigurationName: DeploymentConfigurationName;
+    public readonly deploymentConfigurationArn: DeploymentConfigurationArn;
+
+    constructor(parent: cdk.Construct, id: string, props: CustomServerDeploymentConfigurationRefProps) {
+        super(parent, id);
+
+        this.deploymentConfigurationName = props.deploymentConfigurationName;
+        this.deploymentConfigurationArn = deploymentConfigurationName2Arn(this.deploymentConfigurationName);
+    }
+}
+
+/**
+ * The possible representations of a healthy-host value -
+ * either percent-based, or an absolute number.
+ */
+export enum HostsRepresentation {
+    NUMBER = 'HOST_COUNT',
+    PERCENTAGE = 'FLEET_PERCENT'
+}
+
+export interface MinimumHealthyHosts {
+    readonly representation: HostsRepresentation;
+
+    readonly value: number;
+}
+
+/**
+ * Construction properties of {@link CustomServerDeploymentConfiguration}.
+ */
+export interface CustomServerDeploymentConfigurationProps {
+    /**
+     * The physical, human-readable name of the Deployment Configuration.
+     *
+     * @default a name will be auto-generated
+     */
+    deploymentConfigurationName?: string;
+
+    /**
+     * The minimum healhty hosts setting for this EC2/on-premise Deployment Configuration.
+     */
+    minimumHealthyHosts: MinimumHealthyHosts;
```

Comment:
I think flatting this into two mutually exclusive properties would make the API more concise and less convoluted. I understand this means it's a "synth-time" error as oppose to compile-time, but it's still cheap.

```ts
new DeploymentConfig(this, 'MyConfig', { minHealthyCount: 100 });
```

or:

```ts
new DeploymentConfig(this, 'MyConfig', { minHealthyPercent: 50 });
```

Reply:
I did it this way because it's more aligned with the CFN way of doing it, which seems important (see above ^^). I don't feel super strongly about it though - if you like the 2 mutually exclusive properties better, I have no problem changing it.

Reply:
We have no goal that our APIs will have the same structure as CFN. In fact, in most cases, the CDK offers an opportunity to simplify, improve discoverability and ergonomics. It's very common when we design L2s to flatten the declarative API in CFN.